packages/gatsby/src/schema/infer/index.js

```diff
@@ -1,49 +1,12 @@
 const report = require(`gatsby-cli/lib/reporter`)
+const { ObjectTypeComposer } = require(`graphql-compose`)
 const { getExampleValue } = require(`./example-value`)
 const {
   addNodeInterface,
   getNodeInterface,
 } = require(`../types/node-interface`)
 const { addInferredFields } = require(`./add-inferred-fields`)
-const getInferConfig = require(`./get-infer-config`)
-
-const addInferredType = ({
-  schemaComposer,
-  typeComposer,
-  nodeStore,
-  typeConflictReporter,
-  typeMapping,
-  parentSpan,
-}) => {
-  const typeName = typeComposer.getTypeName()
-  const nodes = nodeStore.getNodesByType(typeName)
-  if (
-    !typeComposer.hasExtension(`plugin`) &&
-    typeComposer.getExtension(`createdFrom`) === `infer`
-  ) {
-    typeComposer.setExtension(`plugin`, nodes[0].internal.owner)
-  }
-  const exampleValue = getExampleValue({
-    nodes,
-    typeName,
-    typeConflictReporter,
-    ignoreFields: [
-      ...getNodeInterface({ schemaComposer }).getFieldNames(),
-      `$loki`,
-    ],
-  })
-
-  addInferredFields({
-    schemaComposer,
-    typeComposer,
-    nodeStore,
-    exampleValue,
-    inferConfig: getInferConfig(typeComposer),
-    typeMapping,
-    parentSpan,
-  })
-  return typeComposer
-}
+const { mergeInferredComposer } = require(`./merge-inferred-composer`)
 
 const addInferredTypes = ({
```

Comment:
Should we rename `addInferredTypes`, now that they don't really add anything? `getNodeTypesToInfer` is a bit clunky maybe but something like that?

Reply:
Good idea.

Reply:
Oh, they do add types again now, so maybe it's okay.